Paws4Me/Paws4Me/Show and Validate Login/View Models/SignInViewModel.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  ViewModels.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/24.
+//
+
+import Foundation
+
+// MARK: - SignInViewModel Delegate
+protocol SignInViewModelDelegate: AnyObject {
+    func successRouting()
+    func show(errorMessage: String)
+}
+
+class SignInViewModel {
+
+    // MARK: - Vars/Lets
+    private weak var delegate: SignInViewModelDelegate?
+
+    init(delegate: SignInViewModelDelegate) {
+        self.delegate = delegate
+    }
+
+    // MARK: - Functions
+    func loginUser(username: String, password: String) {
+        if username.isEmpty || password.isEmpty {
+            delegate?.show(errorMessage: "Please fill in your username and password")
+        }
+        let bLoggedIn = isValidCredentials(username: username, password: password)
+
+        if bLoggedIn {
```

Comment:
you could you the ternary operator here 😄 

Also don't prefix this variable name with "b"

Reply:
Thank you. This is much cleaner 👍🏻